libchisel/src/utils.rs

```diff
@@ -0,0 +1,33 @@
+//! These are helpers to be used internally.
+
+use parity_wasm::elements::{deserialize_buffer, serialize, Module};
+
+pub trait HasNamesSection {
+    /// Returns true if the module has a NamesSection.
+    fn has_names_section(&self) -> bool;
+}
+
+impl HasNamesSection for Module {
+    fn has_names_section(&self) -> bool {
+        // Lets forcefully parse in case it wasn't yet.
+        let mut module = self.clone().parse_names().unwrap();
+        module.names_section().is_some()
+    }
+}
+
+pub trait SerializationHelpers {
+    /// Deserialize bytecode to a Module.
+    fn from_slice(input: &[u8]) -> Module;
```

Comment:
Are you sure we would rather panic internally rather than return a `Result<T>`?

Reply:
That was considered as an option, but this right now is only using in testing code, so its fine.

Reply:
ok